packages/fxa-content-server/app/scripts/lib/survey-filter.js

```diff
@@ -0,0 +1,236 @@
+
+import UserAgent from './user-agent';
+
+// All but the default export here is to make testing easier.
+
+const falseFn = () => false;
+export const NONE = Symbol();
+
+export const participatedRecently = (
+  previousParticipationTime,
+  doNotBotherSpan
+) => {
+  if (
+    previousParticipationTime === null ||
+    previousParticipationTime === undefined
+  ) {
+    return false;
+  }
+  return Date.now() - previousParticipationTime < doNotBotherSpan;
+};
+
+export const getConditionWithKey = (conditions, key) => {
+  // It is not something we need to check.
+  if (!Object.keys(conditions).includes(key)) {
+    return NONE;
+  }
+
+  return conditions[key];
+};
+
+export const createConditionCheckFn = (valSourceCheck) => (comparator) => (
+  key
+) => (conds, valSource) => {
+  // This turned out a little janky.  But we want to return true early if the
+  // condition is not in the configs.
+  const condVal = getConditionWithKey(conds, key);
+  if (condVal === NONE) {
+    return true;
+  }
+  return valSourceCheck(valSource)(comparator)(condVal);
+};
+
+export const createAsyncConditionCheckFn = (valSourceCheck) => (comparator) => (
+  key
+) => async (conds, valSource) => {
+  const condVal = getConditionWithKey(conds, key);
+  if (condVal === NONE) {
+    return true;
+  }
+  return (await valSourceCheck(valSource)(comparator))(condVal);
+};
+
+let ua;
+export const checkConditionInUa = (window) => (checkUa) => {
+  if (!window || !window.navigator || !window.navigator.userAgent) {
+    return falseFn;
+  }
+
+  // @TODO since sinon cannot spy on ES modules, it's difficult to test that
+  // the ua string parsing happens only once.  Consider passing a parser
+  // function as dependency.
+
+  if (!ua) {
+    ua = new UserAgent(window.navigator.userAgent);
+  }
+
+  return checkUa(ua);
+};
+
+const createUaConditionCheckFn = createConditionCheckFn(checkConditionInUa);
+
+export const checkUaDeviceType = (ua) => (val) => {
+  return ua.genericDeviceType().toLowerCase() === val.toLowerCase();
+};
+
+export const checkUaOsName = (ua) => (val) =>
+  ua.os.name.toLowerCase() === val.toLowerCase();
+
+export const checkUaBrowser = (ua) => (val) =>
+  ua.browser.name.toLowerCase() === val.toLowerCase();
+
+// Ref: https://github.com/mozilla/fxa/blob/9b2d9d1/packages/fxa-content-server/app/scripts/lib/user-agent.js#L182
+export const hasDesiredDeviceType = createUaConditionCheckFn(checkUaDeviceType)(
+  'deviceType'
+);
+export const hasDesiredOs = createUaConditionCheckFn(checkUaOsName)('os');
+export const hasDesiredBrowser = createUaConditionCheckFn(checkUaBrowser)(
+  'browser'
+);
+
+export const userAgentChecks = (conds, win) =>
+  hasDesiredBrowser(conds, win) &&
+  hasDesiredDeviceType(conds, win) &&
+  hasDesiredOs(conds, win);
+
+const applyWithSourceVal = (sourceVal) => (fn) => fn(sourceVal);
+
+export const checkRelierClientId = (relier) => (val) => {
+  return relier.get('clientId') === val;
+};
```

Comment:
nit: we don't need the `{}` and `return`.  There's only one expression, so the shorthand `a => b => c => relier.get('clientId') === val;` would work.

Reply:
I added the {} and return so I could add a debugger statement and for got to fix it back, will fix :)